pennylane/tape/tapes/qubit_param_shift.py

```diff
@@ -169,7 +169,7 @@ def processing_fn(results):
                 array[float]: 1-dimensional array of length determined by the tape output
                 measurement statistics
             """
-            return np.dot(coeffs, results)
+            return np.dot(coeffs, np.squeeze(results))

```

Comment:
How does this change affect the [hotfix in the tape QNode](https://github.com/PennyLaneAI/pennylane/blob/a2fa39cd1c21725e0f1fe1e9361ef856c4d990de/pennylane/tape/qnode.py#L433)? Would that still be required?

Reply:
Yep, that will still be required. Note that the `squeeze` here is functionally independent of the one in the QNode.

* The one in `QNode.__call__` is designed so that the QNode output matches the users `return` function. E.g., if the user specifies `return qml.expval(qml.PauliZ(0))`, squeezing occurs so that a float is returned. If instead the user specifies `return [qml.expval(qml.PauliZ(0))]` or `return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliX(0))`, then no squeezing occurs and a list and/or tuple is returned.

* The squeezing here is low level, and not connected to the QNode return statement. Here, we are simply ensuring that `np.dot()` behaves as we need it to.